src/components/InstalledApp.vue

```diff
@@ -166,7 +166,7 @@ export default {
         if (this.torOnly) {
           return "#";
         }
-        return `http://${window.location.hostname}:${this.app.port}${this.app.path}`;
+        return `${window.location.origin}:${this.app.port}${this.app.path}`;
```

Comment:
We're swapping out `http://${window.location.hostname}` for `${window.location.origin}` here but the two are not actually equivalent. `origin` includes the port, `hostname` doesn't so if the dashboard is run on a non-standard port it looks like these urls would be incorrect.

e.g If the dashboard was hosted on `http://umbrel.local:1234` and an app was on port 3000 then current code would link to `http://umbrel.local:3000` but after this PR it would link to `http://umbrel.local:1234:3000`

If we instead keep the original logic but just swap out the hardcoded `http:` protocol for the current protocol dynamically with `window.location.protocol` it should have the desired result:

```suggestion
        return `${window.location.protocol}//${window.location.hostname}:${this.app.port}${this.app.path}`;
```